### ISSUE TITLE
Fork loop issue

### ISSUE THREAD

**Issue by rollecode** [coder8338/appbox_installer#55]
Recently I noticed I cannot log in to my box and after boot got `-bash: fork: retry: Resource temporarily unavailable`. After investigation with service provider it seemed I had thousands of zombie processes:

```
18104000    2003  0.0  0.0      0     0 ?        Zs   Oct02   0:00 [s6-supervise] <defunct>
18104000    2275  0.0  0.0      0     0 ?        Zs   00:41   0:00 [s6-supervise] <defunct>
18104000    3413  0.0  0.0      0     0 ?        Zs   Oct02   0:00 [s6-supervise] <defunct>
18104000    3434  0.0  0.0      0     0 ?        Zs   Oct02   0:00 [s6-supervise] <defunct>
18104000    4380  0.0  0.0      0     0 ?        Zs   09:10   0:00 [s6-supervise] <defunct>
18104000    5777  0.0  0.0      0     0 ?        Zs   00:41   0:00 [s6-supervise] <defunct>
18104000    8202  0.0  0.0      0     0 ?        Zs   Oct02   0:00 [s6-supervise] <defunct>
18104000    8215  0.0  0.0      0     0 ?        Zs   Oct02   0:00 [s6-supervise] <defunct>
18104000    8253  0.0  0.0      0     0 ?        Zs   Oct02   0:00 [s6-supervise] <defunct>
18104000    8482  0.0  0.0      0     0 ?        Zs   Oct02   0:00 [s6-supervise] <defunct>
18104000   10737  0.0  0.0      0     0 ?        Zs   Oct02   0:00 [s6-supervise] <defunct>
18104000   10772  0.0  0.0      0     0 ?        Zs   Oct02   0:00 [s6-supervise] <defunct>
18104000   11146  0.0  0.0      0     0 ?        Zs   00:41   0:00 [s6-supervise] <defunct>
18104000   13183  0.0  0.0      0     0 ?        Zs   Oct02   0:00 [s6-supervise] <defunct>
18104000   13228  0.0  0.0      0     0 ?        Zs   Oct02   0:00 [s6-supervise] <defunct>
18104000   14107  0.0  0.0      0     0 ?        Zs   Oct02   0:00 [s6-supervise] <defunct>
18104000   14187  0.0  0.0      0     0 ?        Zs   Oct02   0:00 [s6-supervise] <defunct>
18104000   15527  0.0  0.0      0     0 ?        Zs   00:41   0:00 [s6-supervise] <defunct>
18104000   16522  0.0  0.0      0     0 ?        Zs   09:11   0:00 [s6-supervise] <defunct>
18104000   17978  0.0  0.0      0     0 ?        Zs   Oct02   0:00 [s6-supervise] <defunct>
18104000   17979  0.0  0.0      0     0 ?        Zs   Oct02   0:00 [s6-supervise] <defunct>
18104000   20842  0.0  0.0      0     0 ?        Zs   00:42   0:00 [s6-supervise] <defunct>
18104000   20953  0.0  0.0      0     0 ?        Zs   Oct02   0:00 [s6-supervise] <defunct>
18104000   20970  0.0  0.0      0     0 ?        Zs   Oct02   0:00 [s6-supervise] <defunct>
18104000   21379  0.0  0.0      0     0 ?        Zs   Oct02   0:00 [s6-supervise] <defunct>
```

I quickly reinstalled sonarr (it could not be found), overseerr and lidarr because I suspected one of those apps caused this (sonarr maybe because I couldn't access to it). After reinstalling them I have not encountered this problem any more.

I never really figured out the root cause but s6-supervise seems to point out that some defunct process has been looping on and on and caused resourcse to run out. Any way to prevent this in the future?

**Comment by rollecode** [comment]
It happened again and I have no idea what causes it. Right now I'm locked out because of this issue, so it's kinda serious. Need to contact the support again...

```
ssh_exchange_identification: Connection closed by remote host
```

**Comment by rollecode** [comment]
This ultimately destroyed the ubuntu app completely... FYI. 🤷

**Comment by coder8338** [comment]
Haha it was your code which I pulled for the updatetool app!

Let me see if I can fix it. Did you test the code before you submitted a PR??

**Comment by rollecode** [comment]
> Haha it was your code which I pulled for the updatetool app!
> 
> Let me see if I can fix it. Did you test the code before you submitted a PR??

Yes, I tested it at it was in use for days on end. I suspect it's related to UpdateTool because it kinda has a schedule function in itself. Or perhaps sonarr app is broken somehow because I could not find it at all when these problems occurred. But I really have no clue was this caused by my changes or something else.

I can no longer test it out because I moved on from appbox, they had other issues as well with my box (like SSL certificate issues with my custom .app TLD domain etc and other connection issues). This fork loop problem might be related to the issues in my box and/or to these scripts. But like I said, I have no idea - I had no way to investigate further because was locked out from my box alltogether and in the end the container got fully corrupted after pids ran out...
